core/renderer/backend/opengl/RenderTargetGL.cpp

```diff
@@ -47,18 +47,19 @@ void RenderTargetGL::update() const {
     if (!_dirtyFlags) return;
     if (!_defaultRenderTarget)
     {
+        if(bitmask::any(_dirtyFlags, TargetBufferFlags::COLOR_ALL))
         {  // color attachments
             GLenum bufs[MAX_COLOR_ATTCHMENT] = {GL_NONE};
             for (size_t i = 0; i < MAX_COLOR_ATTCHMENT; ++i)
             {
+                bufs[i] = GL_COLOR_ATTACHMENT0 + i;
```

Comment:
This has to be set only if there's an attachment, otherwise it can trigger the same issue that @solan-solan was having.

Reply:
There is full code of RenderTargetGL::update depend on @halx99 first commit

Definition of RenderTargetGL::_bufs:
    `mutable GLenum _bufs[MAX_COLOR_ATTCHMENT] = { GL_NONE };`

```
void RenderTargetGL::update() const {
    if (!_dirtyFlags) return;
    if (!_defaultRenderTarget)
    {
        {  // color attachments
            for (size_t i = 0; i < MAX_COLOR_ATTCHMENT; ++i)
            {
                if (!bitmask::any(_dirtyFlags, getMRTColorFlag(i)))
                    continue;
                auto textureInfo = _color[i];
                GLuint textureHandler = static_cast<GLuint>(0);
                _bufs[i] = GL_NONE;
                if (textureInfo.texture != nullptr)
                {
                    textureHandler = static_cast<GLuint>(textureInfo.texture->getHandler());
                    _bufs[i] = GL_COLOR_ATTACHMENT0 + i;
                }
                glFramebufferTexture2D(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0 + i, GL_TEXTURE_2D, textureHandler,
                                       textureInfo.level);
            }
#if AX_TARGET_PLATFORM == AX_PLATFORM_WIN32 || AX_TARGET_PLATFORM == AX_PLATFORM_LINUX
            glDrawBuffers(MAX_COLOR_ATTCHMENT, _bufs);
#endif
            CHECK_GL_ERROR_DEBUG();
        }

        if (bitmask::any(_dirtyFlags, TargetBufferFlags::DEPTH)) {
            // depth attacmhemt
            glFramebufferTexture2D(GL_FRAMEBUFFER, GL_DEPTH_ATTACHMENT, GL_TEXTURE_2D,
                                   _depth.texture != nullptr ? _depth.texture->getHandler() : 0, _depth.level);
            CHECK_GL_ERROR_DEBUG();
        }

        if (bitmask::any(_dirtyFlags, TargetBufferFlags::STENCIL))
        {
            // stencil attachment
            glFramebufferTexture2D(GL_FRAMEBUFFER, GL_STENCIL_ATTACHMENT, GL_TEXTURE_2D,
                                   _stencil.texture != nullptr ? _stencil.texture->getHandler() : 0, _stencil.level);
            CHECK_GL_ERROR_DEBUG();
        }
    }

    _dirtyFlags = TargetBufferFlags::NONE;
}
```

Describe please case where it can break if it is, since it works for me.